api-generation/src/main/scala/ai/h2o/sparkling/api/generation/Runner.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.api.generation
+
+import java.io.{File, PrintWriter}
+
+import ai.h2o.sparkling.api.generation.common._
+import ai.h2o.sparkling.api.generation.scala.{AlgorithmTemplate, ParametersTemplate}
+import ai.h2o.sparkling.utils.ScalaUtils._
+import hex.deeplearning.DeepLearningModel.DeepLearningParameters
+import hex.schemas._
+import hex.tree.gbm.GBMModel.GBMParameters
+import hex.glm.GLMModel.GLMParameters
+import hex.kmeans.KMeansModel.KMeansParameters
+import hex.tree.drf.DRFModel.DRFParameters
+import hex.tree.xgboost.XGBoostModel.XGBoostParameters
+
+object Runner {
+
+  private val explicitDefaultValues = Map("max_w2" -> "java.lang.Float.MAX_VALUE")
+
+  private def parametersConfiguration: Seq[ParameterSubstitutionContext] = {
+    val monotonicity = ExplicitField("monotone_constraints", "HasMonotoneConstraints")
```

Comment:
Could we do classOf[HasMonotoneConstraints].getSimpleName?

Reply:
IMHO, it's not possible. To do that, the `api-generation` project would need to reference the `ml` project. Since `ml` references `api-generation` already. We would get a cyclic reference.

Reply:
Oki, thanks for explanation!